crates/release_plz_core/src/git/backend.rs

```diff
@@ -540,49 +563,163 @@ impl GitClient {
-                self.client
-                    .post(format!("{}/{}/labels", self.issues_url(), pr_number))
-                    .json(&json!({
-                        "labels": pr.labels
-                    }))
-                    .send()
-                    .await?
-                    .successful_status()
+            BackendType::Github => self.post_github_labels(labels, pr_number).await,
+            BackendType::Gitlab => self.post_gitlab_labels(labels, pr_number).await,
+            BackendType::Gitea => {
+                let (labels_to_create, mut label_ids) = self
+                    .get_pr_info_and_categorize_labels(pr_number, labels)
                     .await?;
 
-                Ok(())
+                let new_label_ids = self.create_labels(&labels_to_create).await?;
+                label_ids.extend(new_label_ids);
+
+                anyhow::ensure!(
+                    !label_ids.is_empty(),
+                    "The provided labels: {labels:?} \n
+                        were not added to PR #{pr_number}",
+                );
+                self.post_gitea_labels(&label_ids, pr_number).await
             }
-            BackendType::Gitlab => {
-                self.client
-                    .put(format!("{}/{}", self.pulls_url(), pr_number))
-                    .json(&json!({
-                        "add_labels": pr.labels.iter().join(",")
-                    }))
-                    .send()
-                    .await?
-                    .successful_status()
-                    .await?;
+        }
+    }
+
+    fn pr_labels_url(&self, pr_number: u64) -> String {
+        format!("{}/{}/labels", self.issues_url(), pr_number)
+    }
+
+    /// Add all labels to PR
+    async fn post_github_labels(&self, labels: &[String], pr_number: u64) -> anyhow::Result<()> {
+        self.client
+            .post(self.pr_labels_url(pr_number))
+            .json(&json!({
+                "labels": labels
+            }))
+            .send()
+            .await?
+            .successful_status()
+            .await?;
+
+        Ok(())
+    }
+
+    /// Add all labels to PR
+    async fn post_gitlab_labels(&self, labels: &[String], pr_number: u64) -> anyhow::Result<()> {
+        self.client
+            .put(format!("{}/{}", self.pulls_url(), pr_number))
+            .json(&json!({
+                "add_labels": labels.iter().join(",")
+            }))
+            .send()
+            .await?
+            .successful_status()
+            .await?;
 
-                Ok(())
+        Ok(())
+    }
+
+    /// Add all labels to PR
+    async fn post_gitea_labels(&self, label_ids: &[u64], pr_number: u64) -> anyhow::Result<()> {
+        self.client
+            .post(self.pr_labels_url(pr_number))
+            .json(&json!({ "labels": label_ids }))
+            .send()
+            .await?
+            .successful_status()
+            .await?;
+        Ok(())
+    }
+
+    async fn get_pr_info_and_categorize_labels(
```

Comment:
can you document the return type of this function?

Reply:
Also, I wonder if we need this function at all.

Does Gitea complaint if we try to add a label that already exists in the issue?
If yes, we should capture the returned status code in post_gitea_labels function and add a context to the error saying that there's a bug in release-plz

Reply:
Adjusting the system to retrieve label information from the repository level.

> Does Gitea give an error if we attempt to add a label that already exists in the issue?

No, it does not handle duplication. I may need to implement a case check when filtering labels for creation to minimize the risk of duplication. This behaviour is similar to that of GitHub.